src/renderers/webgl/WebGLRenderContexts.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @author Mugen87 / https://github.com/Mugen87
+ */
+
+import { WebGLLights } from './WebGLLights.js';
+
+var count = 0;
+
+function WebGLRenderContext() {
```

Comment:
Definitely, don't call this `WebGLRenderContext` due to ambiguity with [`WebGLRenderingContext`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext)

I don't have a better name because this also does things with sprites. How would you categorize the notion of lights, shadows, and sprites?

Say hypothetically if we went with `WebGLRenderLight`, then change the name of `WebGLRenderContexts` to `WebGLRenderLightCollection`.

Reply:
> How would you categorize the notion of lights, shadows, and sprites?

This class could manage even more stuff that needs to be preserved for a single render call. So it would be great to have a generic name. Too bad that "context" is already reserved. What about `WebGLRenderState`?

Reply:
There is an overlap with `WebGLState `, but we also have `WebGLBufferRenderer` next to `WebGLRenderer`...

Reply:
A `State` would implicate something that has a case, thus something that can be `switched`.

If all this does is preserve a collection of elements that can be rendered, then it should be called like that. What do you think about `WebGLRenderSet`?

Reply:
> A State would implicate something that has a case, thus something that can be switched.

Whenever we change the reference of `currentRenderState`, we actually perform some sort of switch to another render state.

> What do you think about WebGLRenderSet?

Maybe an option. Let's see what the others think about the names (and the entire approach^^).